isis/src/base/objs/StringBlob/StringBlob.h

```diff
@@ -25,24 +28,31 @@ namespace Isis {
    * @internal
    *   @history 2020-11-19 Kristin Berry - Original Version
     */
-  class StringBlob : public Isis::Blob {
+  class StringBlob {
     public:
       StringBlob();
-      StringBlob(const QString &file);
+      StringBlob(Blob &blob);
       StringBlob(std::string str, QString name);
-      ~StringBlob();
+      virtual ~StringBlob();
+
+      virtual Blob *toBlob() const;
 
-      std::string string() {
+      std::string string() const {
         return m_string;
       }
 
-    protected:
-      // prepare data for writing
-      void WriteInit();
-      void ReadData(std::istream &stream);
+      QString name() const {
+        return m_name;
```

Comment:
These are generally all over the place in ISIS but I would rather these be getString and getName